### PR TITLE
fix(range): setting dir on ion-range to enable rtl mode now supported

### PR DESCRIPTION
**Pull request by sean-perkins**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

`ion-range` only supports RTL when the direction is set at the document level.

Issue Number: #20201, Tech Debt

## What is the new behavior?

- `ion-range` will respect `dir` specified on the host element (overrides the document value)
- `ion-range` uses the `isRTL` utility

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by amandaesmith3 — APPROVED** [review]

File: core/src/components/range/range.tsx [hunk 1]
```diff
@@ -432,10 +433,8 @@ export class Range implements ComponentInterface {
     const barStart = `${ratioLower * 100}%`;
     const barEnd = `${100 - ratioUpper * 100}%`;
 
-    const doc = document;
-    const isRTL = doc.dir === 'rtl';
-    const start = isRTL ? 'right' : 'left';
-    const end = isRTL ? 'left' : 'right';
+    const start = isRTL(this.el) ? 'right' : 'left';
+    const end = isRTL(this.el) ? 'left' : 'right';
```
Comment (liamdebeasi) [inline core/src/components/range/range.tsx:437]: Minor optimization, but we can cache `isRTL(this.el)` to a variable so we don't need to call it twice.

File: core/src/components/range/range.tsx [hunk 2]
```diff
@@ -502,7 +501,7 @@ export class Range implements ComponentInterface {
             part="bar-active"
           />
 
-          { renderKnob(isRTL, {
+          {renderKnob(isRTL(this.el), {
```
Comment (liamdebeasi) [inline core/src/components/range/range.tsx:504]: Same thing here. The calls in this function can cached within the function context

**Review by liamdebeasi — APPROVED** [review]
Two minor comments, but otherwise looks good. Let's make sure the commit subject follows the guidelines too. Maybe something like `fix(range): setting dir on ion-range to enable rtl mode now supported`?

**Comment by liamdebeasi** [comment]
Can we also check to see if this is fixed? https://github.com/ionic-team/ionic-framework/issues/20201

**Comment by sean-perkins** [comment]
> Can we also check to see if this is fixed? #20201

@liamdebeasi yeah, that issue will be resolved after I add the utility check for one more area I missed 😬

**Comment by liamdebeasi** [comment]
Nice!!

**Comment by Ornel21** [comment]
npm run build

**Pull request merged**